libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -178,13 +181,15 @@ internal object MapboxNavigationTelemetry {
     private val navigationSessionStateObserver = NavigationSessionStateObserver { sessionState ->
         log("session state is $sessionState")
```

Comment:
```suggestion
        log("session state is $sessionState")
        this.sessionState = sessionState
```
Otherwise, some of the methods below would use the old, outdated field value. Would be great to cover this with a unit test as well.

Reply:
Clarified that this setup is correct for the old schema, due to be updated in https://github.com/mapbox/mapbox-navigation-android/pull/4693.